src/Language/PureScript/Errors.hs

```diff
@@ -1409,6 +1409,8 @@ toTypelevelString (TypeApp (TypeConstructor f) x)
   | f == primSubName C.typeError "Text" = toTypelevelString x
 toTypelevelString (TypeApp (TypeConstructor f) x)
   | f == primSubName C.typeError "Quote" = Just (typeAsBox x)
+toTypelevelString (TypeApp (TypeConstructor f) x)
+  | f == primSubName C.typeError "QuoteLabel" = Just (typeAsBox x)
```

Comment:
I think for this, you don't want to use `typeAsBox`. You want to match directly on `TypelevelString` for `x`, along with `Just . line . prettyPrintLabel $ x`.